src/device/r4300/r4300_core.h

```diff
@@ -158,6 +167,25 @@ struct r4300_core
 #endif
 #endif
 
+    /* Memory accesses variables */
+    uint64_t* rdword;
+
+#if NEW_DYNAREC != NEW_DYNAREC_ARM
+/* ARM dynarec uses a different memory layout */
+    uint32_t wmask;
+
+    union {
+        uint32_t wword;
+        uint64_t wdword;
+    };
+
+    uint32_t address;
+#endif
+
+#if NEW_DYNAREC == NEW_DYNAREC_ARM
+    char extra_memory[33554432];
```

Comment:
Taken from new_dynarec.mediawiki

Branch instructions are limited to a +/-32MB range on ARM.  In some cases, the dynamic recompiler needs to generate calls to locations beyond this range.  This is accomplished via a jump table located at the end of the code generation area, and the full address is loaded via a pointer.  The jump table is generated in arch_init().

As these indirect jumps cause some delay, it is best to avoid this situation if possible, by locating this area close to the other executable code.